packages/web/src/features/translation/TranslationProgressBar.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useRef, useState } from 'react';
+import { useTextWidth } from '../../shared/hooks/useTextWidth';
+import { useTranslation } from 'react-i18next';
+
+interface TranslationProgressBarProps {
+  approvedCount: number;
+  wordCount: number;
+}
+
+export default function TranslationProgressBar(
+  props: TranslationProgressBarProps
+) {
+  const { t } = useTranslation();
+  const progressElementRef = useRef<HTMLDivElement>(null);
+
+  const percentageFull = (props.approvedCount / props.wordCount) * 100;
+
+  const progressText = `${props.approvedCount}/${props.wordCount} ${t(
+    'translate:words'
+  )} (${percentageFull.toFixed(1)}%)`;
+  const textElementWidth =
+    32 + // 32px for the left/start margin
+    useTextWidth({
+      text: progressText,
+      fontSize: '12px',
+      fontFamily: 'inherit',
+    }) +
+    12; // 12px for the right/end margin;
+
+  const [textFitsInside, setTextFitsInside] = useState(true);
+
+  useEffect(() => {
+    const { current: progressElement } = progressElementRef;
+    if (!progressElement) return;
+    const resizeObserver = new ResizeObserver(() => {
+      setTextFitsInside(textElementWidth <= progressElement.offsetWidth);
+    });
```

Comment:
We should create the resize observer with useRef, and then observe in the effect and unobserve when the effect is torn down. This prevents the observer from being recreated whenever the effect runs

Reply:
Done.

Reply:
I ended up also adding up using `useCallback` to capture the value of `textElementWidth`